src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32.cs

```diff
@@ -444,6 +444,36 @@ private static Win32 GetProgramFromPath(string path)
             return entry;
         }
 
+        public class removeDuplicatesComparer : IEqualityComparer<Win32>
+        {
+            public bool Equals(Win32 app1, Win32 app2)
+            {
+                
+                if(!string.IsNullOrEmpty(app1.Name) && !string.IsNullOrEmpty(app2.Name)
+                    && !string.IsNullOrEmpty(app1.ExecutableName) && !string.IsNullOrEmpty(app2.ExecutableName)
+                    && !string.IsNullOrEmpty(app1.FullPath) && !string.IsNullOrEmpty(app2.FullPath))
+                {
+                    return app1.Name.Equals(app2.Name, StringComparison.OrdinalIgnoreCase) 
+                        && app1.ExecutableName.Equals(app2.ExecutableName, StringComparison.OrdinalIgnoreCase) 
+                        && app1.FullPath.Equals(app2.FullPath, StringComparison.OrdinalIgnoreCase);
+                }
+                return false;
+            }
+
+            public int GetHashCode(Win32 obj)
+            {
+                return obj.FullPath.GetHashCode();

```

Comment:
The equal functions do take all three variables into account but hashing doesn't. So comparison using hashing and equal would produce different results.

Reply:
Good catch! Used exor.

Reply:
I don't think exor will ensure uniqueness for hashing. Swapping name and executable name will give false on using `Equal` but true on comparing hash codes because exor is commutative operation. 

You can use something like below snippet which has a very low probability of being equal(which seems to be standard practice as per [this](https://stackoverflow.com/questions/2730865/how-do-i-calculate-a-good-hash-code-for-a-list-of-strings) post): 
```
final int prime = 31;
int result = 1;
for( String s : strings )
{
    result = result * prime + s.hashCode();
}
```

Reply:
modified it as you suggested.